### PR TITLE
add module for AWS

### LINKED ISSUE #933: Needs add new module file to compile on AWS
We are working on port EMC global-workflow to AWS.
To make start with, we need to add a new module file to load modules on AWS.

### PR DESCRIPTION
**Pull request by weihuang-jedi**
We are working on port EMC global-workflow to AWS.
To make start with, we need to add a new module file to load modules on AWS.

This PR is to solve issue #933.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi Did you test your PR on AWS?

File: modulefiles/noaacloud.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
+help([[
+Load environment to build UPP on hera
```
Comment (aerorahul) [inline modulefiles/noaacloud.lua]:
```suggestion
Load environment to build UPP on NOAA Cloud
```

Comment (weihuang-jedi) [reply modulefiles/noaacloud.lua]: Nice catch. Updated.

Compiled on AWS:

[Wei.Huang@NOAA-AWS:/contrib/Wei.Huang/dev/upp]$ tests/compile_upp.sh
Building for machine noaacloud, compiler intel

Currently Loaded Modules:
  1) intel/2021.3.0                    8) libjpeg/2.1.0      15) zstd/1.5.2              22) crtm-fix/2.4.0.1_emc  29) w3emc/2.10.0
  2) stack-intel/2021.3.0              9) jasper/2.0.32      16) c-blosc/1.21.5          23) git-lfs/2.4.1         30) nemsio/2.5.4
  3) impi/2021.3.0                    10) zlib/1.2.13        17) netcdf-c/4.9.2          24) crtm/2.4.0.1          31) sigio/2.3.2
  4) stack-intel-oneapi-mpi/2021.3.0  11) libpng/1.6.37      18) netcdf-fortran/4.6.1    25) g2/3.4.5              32) sfcio/1.4.1
  5) nghttp2/1.57.0                   12) pkg-config/0.27.1  19) parallel-netcdf/1.12.2  26) g2tmpl/1.10.2         33) wrf-io/1.2.0
  6) curl/8.4.0                       13) hdf5/1.14.0        20) parallelio/2.5.10       27) ip/4.3.0              34) upp_common
  7) cmake/3.23.1                     14) snappy/1.1.10      21) bacio/2.4.1             28) sp/2.5.0              35) noaacloud



+ BUILD_DIR=build
+ rm -rf build install
+ mkdir -p build
+ cd build

......

-- Installing: /contrib/Wei.Huang/dev/upp/install/lib/cmake/upp/upp-targets.cmake
-- Installing: /contrib/Wei.Huang/dev/upp/install/lib/cmake/upp/upp-targets-release.cmake
+ rm -rf /contrib/Wei.Huang/dev/upp/exec
+ mkdir -p /contrib/Wei.Huang/dev/upp/exec
+ cp ../install/bin/upp.x /contrib/Wei.Huang/dev/upp/exec/.

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi Can you sync your branch with the latest UPP develop?

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by weihuang-jedi** [comment]
Synced the fork with origin.

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi Can you sync your branch for PR processing?

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi It seems to me #930, #932 and #937 were not synced yet. You might refer to the sync procedure [here](https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#contribute-changes-to-the-upp-main-development-branch-develop)

**Comment by weihuang-jedi** [comment]
I tried click "sync fork" button, and got:

upp fork

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi Please check your work branch 'wei-epic-aws' instead of 'develop'.

**Comment by weihuang-jedi** [comment]
I guess I missed the merge to my branch part.

here is what I did:

  173  git clone ***@***.***:NOAA-EPIC/UPP-cloud.git upp-cloud
  174  cd upp-cloud
  175  git pull
  176  git br
  177  git co wei-epic-aws
  178  git merge origin wei-epic-aws
  179  git push origin wei-epic-aws

  185  tests/compile_upp.sh

The compile was successful.

On hera, did the "git pull" on wei-epic-aws branch, do see few files
updated.
Compile was successful.

Sorry for those mistakes, and mis-communication.

Wei

**Comment by WenMeng-NOAA** [comment]
@weihuang-jedi Now your branch looks good to me. Thanks for syncing.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs were completed on WCOSS2. There will be no baseline update in this PR. This PR is ready for merging.

**Review by WenMeng-NOAA — APPROVED** [review]

**Pull request merged**